src/java/org/apache/cassandra/db/streaming/FileStreamMetricsListener.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.streaming;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.codahale.metrics.Counter;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.metrics.StreamingMetrics;
+import org.apache.cassandra.streaming.ProgressInfo;
+import org.apache.cassandra.utils.FBUtilities;
+
+/**
+ * Used to keep track of the bytes streamed for a given file as bytes of the file are transferred and report them to StreamingMetrics.
+ */
+public class FileStreamMetricsListener
+{
+    public static final FileStreamMetricsListener NO_OP = new FileStreamMetricsListener(0L, FBUtilities.getLocalAddressAndPort(), ProgressInfo.Direction.IN){
```

Comment:
Agree that is much cleaner.

Reply:
I was wondering if we should make `FileStreamMetricsListener NO_OP` an explicit testing method instead of a static field (ie. `static FileStreamMetricsListener forTesting()` ) to indicate this is only valid for testing and should not be used by db code - wdyt?

Reply:
It should be package protected and annotation `@VisibleForTesting` should be added on that method.

However, creating a method for that seems to be like unnecessary exercise. Nothing wrong with having it like it is, package protected and annotated as we usually do. That serves the same purpose as having "forTesting()" method.

Reply:
SGTM 👍

Reply:
hmm actually some tests outside the package are using `NO_OP`, so it will probably need to remain public.

Reply:
ah right, that is fine if we remain to have that annotation on it I guess?